pydantic/decorators.py

```diff
@@ -116,21 +117,51 @@ def __call__(self, __cls: Any, __values: _decorators.RootValidatorValues) -> _de
 )
 
 
+# Use a single overload here to get a different signature that requires at least one field be specified
+@overload  # type: ignore[misc]
 def validator(
     __field: str,
     *fields: str,
     pre: bool = False,
+    each_item: bool = False,
+    always: bool = False,
     check_fields: bool | None = None,
     allow_reuse: bool = False,
+) -> Callable[[_V1ValidatorType], _V1ValidatorType]:
+    ...
+
+
+def validator(
+    *fields: str,
+    pre: bool = False,
     each_item: bool = False,
+    always: bool = False,
+    check_fields: bool | None = None,
+    allow_reuse: bool = False,
 ) -> Callable[[_V1ValidatorType], _V1ValidatorType]:
     """
     Decorate methods on the class indicating that they should be used to validate fields
     :param fields: which field(s) the method should be called on
+    :param pre: whether or not this validator should be called before the standard validators (else after)
+    :param each_item: for complex objects (sets, lists etc.) whether to validate individual elements rather than the
+      whole object
+    :param always: whether this method and other validators should be called even if the value is missing
     :param check_fields: whether to check that the fields actually exist on the model
     :param allow_reuse: whether to track and raise an error if another validator refers to the decorated function
     """
-    fields = tuple((__field, *fields))
+    if not fields:
+        raise TypeError('validator with no fields specified')
+    elif isinstance(fields[0], FunctionType):
+        raise TypeError(
+            'validators should be used with fields and keyword arguments, not bare. '
+            "E.g. usage should be `@validator('<field_name>', ...)`"
+        )
+    elif not all(isinstance(field, str) for field in fields):
```

Comment:
The other ones I agree with, this one I think is overkill

Reply:
if we drop the check that the positional items being passed to @field_validator are of type str, this test:
```python
        class Model(BaseModel):
            a: str
            b: str

            @field_validator(['a', 'b'])
            def check_fields(cls, v):
                return v
```
goes from raising: `TypeError: field_validator fields should be passed as separate string args.`
to raising: `TypeError: unhashable type: 'list'`

The docstring of the test (`test_field_validator_bad_fields_throws_configerror`) explicitly says the goal was to raise a custom error in the case of passing a list of str instead of multiple string args. (Admittedly, we don't do that now, but the plan would be to make it raise a pydantic-specific error in those cases as part of a separate PR.) Even given that, do you want to drop this error case?